pkg/scheduler/core/extender.go

```diff
@@ -424,28 +425,22 @@ func (h *HTTPExtender) IsInterested(pod *v1.Pod) bool {
 	if h.managedResources.Len() == 0 {
 		return true
 	}
-	if h.hasManagedResources(pod.Spec.Containers) {
-		return true
-	}
-	if h.hasManagedResources(pod.Spec.InitContainers) {
-		return true
-	}
-	return false
-}
 
-func (h *HTTPExtender) hasManagedResources(containers []v1.Container) bool {
-	for i := range containers {
-		container := &containers[i]
-		for resourceName := range container.Resources.Requests {
+	var interested bool
+	podutil.VisitContainers(&pod.Spec, func(c *v1.Container) bool {
```

Comment:
Is it necessary to consider the ephemeral containers which do not affect scheduling decisions in the extender? I'm not quite sure about this even though the ephemeral containers won't allow defining resources.

Besides the `VisitContainers` use `utilfeature.DefaultFeatureGate.Enabled(features.EphemeralContainers)` which I recall that @bsalamat  has mentioned this could introduce [performance degradation](https://github.com/kubernetes/kubernetes/pull/78319#discussion_r294568048) in the critical scheduling path.

See also https://github.com/kubernetes/kubernetes/pull/81716#discussion_r316106951

> [Some fields of v1.Container imply consequences for the entire pod. For example, one would expect setting resources to increase resources allocated to the pod, but this is not yet supported. We will disallow resources in Ephemeral Containers.](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/20190212-ephemeral-containers.md#updating-a-pod)
>
> [There are no guaranteed resources for ad-hoc troubleshooting. If troubleshooting causes a pod to exceed its resource limit it may be evicted.](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/20190212-ephemeral-containers.md#implementation-detailsnotesconstraints)

Reply:
Hi @draveness 

Nope, it's not necessary, and I'm torn myself on whether it's worth it. I wrote some of my thoughts in https://github.com/kubernetes/kubernetes/pull/79176#issuecomment-523848648. If you have any thoughts, please do add them. If we reach a consensus I'll suggest updating the docs for VisitContainers.

Regardless of the broader issue, it's certainly premature to risk a performance regression. This is a requirement I hadn't considered.

I'll close this PR. Do you want to keep the unit test?

Reply:
> I'll close this PR. Do you want to keep the unit test?

SG, we could keep it and thanks for your work

Reply:
Ok, updated PR to contain only the test.